sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -115,28 +118,48 @@ public HttpResponse processSync(HttpPipelineCallContext context, HttpPipelineNex
     }
 
 
-    private Mono<HttpResponse> attemptAsync(final HttpPipelineCallContext context, final HttpPipelineNextPolicy next,
-        final HttpRequest originalHttpRequest, final int tryCount, final List<Throwable> suppressed) {
-        context.setHttpRequest(originalHttpRequest.copy());
-        context.setData(HttpLoggingPolicy.RETRY_COUNT_CONTEXT, tryCount + 1);
-        return next.clone().process()
-            .flatMap(httpResponse -> {
-                if (shouldRetry(retryStrategy, httpResponse, tryCount)) {
-                    final Duration delayDuration = determineDelayDuration(httpResponse, tryCount, retryStrategy,
-                        retryAfterHeader, retryAfterTimeUnit);
-                    logRetry(tryCount, delayDuration);
+    private Mono<HttpResponse> attemptAsync(HttpPipelineCallContext context, HttpPipelineNextPolicy next,
+        HttpRequest originalHttpRequest, int tryCount, List<Throwable> suppressed) {
+        Mono<HttpResponse> request;
+        if (retryStrategy.getMaxRetries() > 0) {
```

Comment:
Just curious, if we check 

`tryCount < retryStrategy.getMaxRetries()`

does that cover the current case and also avoid need for copying the body for the "last try"? (because there is not going to be another try hence use the original one and drain it)

(Ignore this, if you think current approach is more safe)

Reply:
This was changed to only buffer the request body once if there are retries available. The buffered body is used in a read-only way which should be safe and will reduce additional buffering.